### PR TITLE
Bb/serj/2/add footer component

### PR DESCRIPTION
**Pull request by serglaska**

### CODE REVIEW

File: src/components/SocialNav.js [hunk 1]
```diff
@@ -1,6 +1,47 @@
 import React from 'react';
-//  /////////////////////////////////////////////////////////////////////////////////////////////////
+// icons
+import { compose, withState } from 'recompose';
+import * as I from '../icons';
+import { SocialIconElement, SocialsWrapper } from './ui';
```
Comment (DmitriyBodlev) [inline src/components/SocialNav.js:8]: // components

File: src/components/ui.js [hunk 2]
```diff
@@ -102,18 +102,25 @@ export const SocialIconElement = styled.a`
...
-    grid-template-columns: max-content max-content;
+export const FooterSection = styled(Flex)`
+  ${createMinWithMediaQuery(GC.FIRST_UI_BREAKPOINT)}
+   {
+    display: flex;
+    flex-wrap: wrap-reverse;
   }
 `;
 
+export const FooterBox = styled.div`
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: make export const FooterNav = styled.nav`

File: src/components/Footer.js [hunk 3]
```diff
@@ -4,40 +4,52 @@ import { withTheme } from 'styled-components';
...
 import SocialNav from './SocialNav';
-import { FooterWrapper } from './ui';
+import { FooterWrapper, FooterSection, FooterLink, FooterBox } from './ui';
 // helpers
 import * as H from '../helpers';
 // ui
 import { Text } from '../ui';
 //  /////////////////////////////////////////////////////////////////////////////////////////////////
 
 export const FooterNavigation = ({ locale }) => (
```
Comment (DmitriyBodlev) [inline src/components/Footer.js:14]: // TODO: With theme

File: src/components/SocialNav.js [hunk 4]
```diff
@@ -1,6 +1,48 @@
 import React from 'react';
-//  /////////////////////////////////////////////////////////////////////////////////////////////////
+// icons
+import { compose, withState } from 'recompose';
```
Comment (DmitriyBodlev) [inline src/components/SocialNav.js:2]: move higher

File: src/components/SocialNav.js [hunk 5]
```diff
@@ -1,6 +1,48 @@
...
+  {
+    iconFunction: I.facebook,
+    link: '#',
+  },
+];
+
+export const SocialNav = () => (
+  <SocialsWrapper>
+    {
+      socialData.map((icon) => (
```
Comment (DmitriyBodlev) [inline src/components/SocialNav.js:43]: (icon, index) => (

File: src/components/SocialNav.js [hunk 6]
```diff
@@ -1,6 +1,48 @@
...
+        <SocialIcon {...icon} />
```
Comment (DmitriyBodlev) [inline src/components/SocialNav.js]: <SocialIcon {...icon} key={index} />

File: src/components/ui.js [hunk 7]
```diff
@@ -9,6 +9,7 @@ import { ifElse } from '../helpers';
 import {
   Box,
   Flex,
+  // Grid, no use
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: delete

File: src/components/ui.js [hunk 8]
```diff
@@ -93,6 +86,42 @@ export const HamburgerBtnWrapper = styled('div')`
...
+export const SocialIconElement = styled.a`
+  margin: 0 10px;
+`;
+
+export const SocialsWrapper = styled.nav`
+  width: 250px;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+ `;
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: delete space

File: src/components/SocialNav.js (hunk 1 above)
Comment (serglaska) [reply src/components/SocialNav.js:8]: +

File: src/components/ui.js (hunk 2 above)
Comment (serglaska) [reply src/components/ui.js]: +

File: src/components/Footer.js (hunk 3 above)
Comment (serglaska) [reply src/components/Footer.js:14]: +

File: src/components/SocialNav.js (hunk 4 above)
Comment (serglaska) [reply src/components/SocialNav.js:2]: +

File: src/components/SocialNav.js (hunk 5 above)
Comment (serglaska) [reply src/components/SocialNav.js:43]: +

File: src/components/SocialNav.js (hunk 6 above)
Comment (serglaska) [reply src/components/SocialNav.js]: +

File: src/components/ui.js (hunk 7 above)
Comment (serglaska) [reply src/components/ui.js]: +

**Review by DmitriyBodlev — APPROVED** [review]

File: src/ui/index.js [hunk 9]
```diff
@@ -78,6 +79,35 @@ export const Box = styled.div`
   ${borderRadius}
 `;
 
+export const Grid = styled.div`
```
Comment (VitaliyAndrushko) [inline src/ui/index.js:82]: do we need flex props on grid (like flexWrap)?

File: src/components/ui.js [hunk 10]
```diff
@@ -93,6 +85,41 @@ export const HamburgerBtnWrapper = styled('div')`
...
+  width: 250px;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+
+export const FooterSection = styled(Flex)`
+  ${createMinWithMediaQuery(GC.FIRST_UI_BREAKPOINT)} {
+    display: flex;
```
Comment (VitaliyAndrushko) [inline src/components/ui.js:112]: it's flex by default

File: src/components/ui.js [hunk 11]
```diff
@@ -93,6 +85,41 @@ export const HamburgerBtnWrapper = styled('div')`
...
+    flex-wrap: wrap-reverse;
+    justify-content: space-between;
+  }
+`;
+
+export const FooterNav = styled.nav`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js:118]: use instead box or flex

File: src/components/ui.js [hunk 12]
```diff
@@ -93,6 +85,41 @@ export const HamburgerBtnWrapper = styled('div')`
...
 `;
 
+export const FooterLink = styled(Box)`
+  &:hover {
+    color: #FBFBFB;
+    cursor: pointer;
+  }
+`;
+
+export const FooterWrapper = styled(Flex)`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js:95]: use Flex directly

File: src/components/SocialNav.js [hunk 13]
```diff
@@ -1,6 +1,50 @@
...
+    iconFunction: I.facebook,
+    link: '#',
+  },
+];
+
+export const SocialNav = () => (
+  <SocialsWrapper>
+    {
+      socialData.map((icon) => (
+        <SocialIcon {...icon} key={genShortId()} />
```
Comment (VitaliyAndrushko) [inline src/components/SocialNav.js:44]: use index us key

File: package.json [hunk 14]
```diff
@@ -20,6 +20,7 @@
     "react-dom": "^16.6.3",
     "react-pose": "^4.0.4",
     "recompose": "^0.30.0",
+    "shortid": "^2.2.14",
```
Comment (VitaliyAndrushko) [inline package.json:23]: are you sure we need shortid in our cases?

**Pull request closed without merging**